installHelpers.js

```diff
@@ -0,0 +1,470 @@
+    }
+    if(semver.lt(results[0].adapt_framework, results[1].adapt_framework)) {
+      updateData.adapt_framework = results[1].adapt_framework;
+    }
+    if(_.isEmpty(updateData)) {
+      return callback();
+    }
+    callback(null, updateData);
+  });
+}
+
+function getFrameworkRoot() {
+  return path.join(configuration.serverRoot, 'temp', configuration.getConfig('masterTenantID'), 'adapt_framework');
+}
+
+/**
+* Checks all releases for the latest to match framework value in config.json
+* Recusion required for pagination.
+*/
+function checkLatestAdaptRepoVersion(repoName, callback) {
+  // used in pagination
+  var nextPage = `https://api.github.com/repos/adaptlearning/${repoName}/releases`;
+
+  var _getReleases = function(done) {
+    request({
+      headers: {
+        'User-Agent': DEFAULT_USER_AGENT ,
+        Authorization: 'token 15e160298d59a7a70ac7895c9766b0802735ac99'
+      },
+      uri: nextPage,
+      method: 'GET'
+    }, done);
+  };
+  var _requestHandler = function(error, response, body) {
+    // we've exceeded the API limit
+    if(response.statusCode === 403 && response.headers['x-ratelimit-remaining'] === '0') {
+      var reqsReset = new Date(response.headers['x-ratelimit-reset']*1000);
+      error = `You have exceeded GitHub's request limit of ${response.headers['x-ratelimit-limit']} requests per hour. Please wait until at least ${reqsReset.toTimeString()} before trying again.`;
+    }
+    else if (response.statusCode !== 200) {
+      error = 'GitubAPI did not respond with a 200 status code.';
+    }
+
+    if (error) {
+      return callback(`Couldn't check latest version of ${repoName}\n${error}`);
+    }
+    nextPage = parseLinkHeader(response.headers.link).next;
+    try {
+      var releases = JSON.parse(body);
+    } catch(e) {
+      return callback(`Failed to parse GitHub release data\n${e}`);
+    }
+    var compatibleRelease;
+    var frameworkVersion = configuration.getConfig('framework');
+    if(!frameworkVersion) {
+      return callback(null, releases[0].tag_name);
+    }
+    async.someSeries(releases, function(release, cb) {
+      var isFullRelease = !release.draft && !release.prerelease;
+      if(isFullRelease && semver.satisfies(release.tag_name, frameworkVersion)) {
+        compatibleRelease = release;
+        return cb(null, true);
+      }
+      cb(null, false);
+    }, function(error, satisfied) {
+      if(!satisfied) {
+        if(nextPage) {
+          return _getReleases(_requestHandler);
+        }
+        error = `Couldn't find any releases compatible with specified framework version (${frameworkVersion}), please check that it is a valid version.`;
+      }
+      if(error) {
+        return callback(error);
+      }
+      callback(error, compatibleRelease.tag_name);
+    });
+  };
+  // start recursion
+  _getReleases(_requestHandler);
+}
+
+// taken from https://gist.github.com/niallo/3109252
+function parseLinkHeader(header) {
+  if (header.length === 0) {
```

Comment:
<details>
<summary>Error running <code>node server</code> (click to expand)</summary>

```console
TypeError: Cannot read property 'length' of undefined
    at parseLinkHeader (D:\adapt_authoring\installHelpers.js:209:13)
    at Request._requestHandler [as _callback] (D:\adapt_authoring\installHelpers.js:172:16)
    at Request.self.callback (D:\adapt_authoring\node_modules\request\request.js:188:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (D:\adapt_authoring\node_modules\request\request.js:1171:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (D:\adapt_authoring\node_modules\request\request.js:1091:12)
    at IncomingMessage.g (events.js:292:16)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickDomainCallback (internal/process/next_tick.js:128:9)
```
</details>